fineract-provider/src/main/java/org/apache/fineract/portfolio/delinquency/service/DelinquencyReadPlatformServiceImpl.java

```diff
@@ -97,4 +107,88 @@ public Collection<LoanDelinquencyTagHistoryData> retrieveDelinquencyRangeHistory
         return mapperLoanDelinquencyTagHistory.map(loanDelinquencyTagData);
     }
 
+    @Override
+    public CollectionData calculateLoanCollectionData(final Long loanId) {
+        final LocalDate businessDate = DateUtils.getBusinessLocalDate();
+        final Loan loan = this.loanRepository.getReferenceById(loanId);
+
+        CollectionData collectionData = CollectionData.template();
+        collectionData.setAvailableDisbursementAmount(loan.getApprovedPrincipal().subtract(loan.getDisbursedAmount()));
+        collectionData.setNextPaymentDueDate(loan.possibleNextRepaymentDate());
+
+        final MonetaryCurrency currency = loan.getCurrency();
+        BigDecimal delinquentAmount = BigDecimal.ZERO;
+        // Overdue Days calculation
+        Long overdueDays = 0L;
+        LocalDate overdueSinceDate = null;
+        final List<LoanTransaction> chargebackTransactions = loan.retreiveListOfTransactionsByType(LoanTransactionType.CHARGEBACK);
+        if (chargebackTransactions.isEmpty()) {
```

Comment:
unnecessary condition. The enhance loop will do nothing if it is empty

Reply:
Removed